portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -47,13 +52,39 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 //        response.setContentType("application/json;");
 //        response.getWriter().println(json);
 
-    Query query = new Query("Memes").addSort("timestamp", SortDirection.DESCENDING);
-
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Memes").addSort("timestamp", SortDirection.DESCENDING);
     PreparedQuery results = datastore.prepare(query);
 
-    response.setContentType("text/html;");
+    if( Boolean.parseBoolean( request.getParameter("aRandomOne") ) == true )
+    {
+        List<Entity> memeList = new ArrayList<Entity>();
+        results.asIterable().forEach(memeList::add);
+        Collections.shuffle(memeList);
 
+        response.setContentType("application/json;");
+        for(Entity aMeme : memeList ){
```

Comment:
You can try Java 8 Stream style:

memList.stream().findFirst().ifPresent(meme -> {
  // turn a single meme to string then print
});